api/src/main/java/org/apache/cloudstack/api/command/user/vm/UpdateVMCmd.java

```diff
@@ -263,13 +264,17 @@ public long getEntityOwnerId() {
     @Override
     public void execute() throws ResourceUnavailableException, InsufficientCapacityException, ServerApiException {
         CallContext.current().setEventDetails("Vm Id: " + this._uuidMgr.getUuid(VirtualMachine.class, getId()));
-        UserVm result = _userVmService.updateVirtualMachine(this);
-        if (result != null){
-            UserVmResponse response = _responseGenerator.createUserVmResponse(getResponseView(), "virtualmachine", result).get(0);
-            response.setResponseName(getCommandName());
-            setResponseObject(response);
-        } else {
-            throw new ServerApiException(ApiErrorCode.INTERNAL_ERROR, "Failed to update vm");
+        try {
+            UserVm result = _userVmService.updateVirtualMachine(this);
+            if (result != null) {
+                UserVmResponse response = _responseGenerator.createUserVmResponse(getResponseView(), "virtualmachine", result).get(0);
+                response.setResponseName(getCommandName());
+                setResponseObject(response);
+            } else {
+                throw new ServerApiException(ApiErrorCode.INTERNAL_ERROR, "Failed to update vm");
+            }
+        } catch (Exception e) {
```

Comment:
this thrown exception will be caught immediately.
can we be more specific in the catch, if not out of pokemon principle, at least to not catch the `ServerApiException`